src/main/kotlin/string/Calculator.kt

```diff
@@ -0,0 +1,58 @@
+package string
+
+class Calculator() {
+    fun sum(input: String?): Int {
+        if (input.isNullOrBlank()) {
+            return 0
+        }
+
+        return add(parse(input))
+    }
+
+    private fun add(inputs: List<String>): Int {
+        var result = 0
+
+        for (input in inputs) {
+            result += input.toInt()
+        }
+
+        return result
+    }
+
+    private fun parse(input: String): List<String> {
+        return validateEach(split(input))
+    }
+
+    private fun split(input: String): List<String> {
+        return customDelimiterSplit(input) ?: input.split(DELIMITER_REGEX.toRegex())
+    }
+
+    private fun customDelimiterSplit(input: String): List<String>? {
+        val result = Regex("//(.)\n(.*)").find(input)
+
+        return result?.let {
+            val customDelimiter = it.groupValues[1]
+            return it.groupValues[2].split(customDelimiter)
+        }
+    }
+
+    private fun validateEach(inputs: List<String>): List<String> {
+        for (input in inputs) {
+            validateEachInput(input)
+        }
+
+        return inputs
+    }
+
+    private fun validateEachInput(input: String) {
+        val num = input.toIntOrNull()
+
+        if (num == null || num < 0) {
+            throw RuntimeException("음수 혹은 숫자가 아닌 값이 들어왔습니다")
+        }
```

Comment:
require() 를 활용해보는 것은 어떨까요?

Reply:
음 require을 만족하지 않았을 경우 illegalargumentexception 예외가 발생하는 걸로 알고 있습니다.
위에서 require을 사용한다면 아래와 같은 코드가 되는 걸까요..?
@malibinYun 
```kotlin
try {
  require(num != null && num >= 0)
} catch (e: illegalargumentexception) {
  throw RuntimeException("음수 혹은 숫자가 아닌 값이 들어왔습니다") 
}
```

Reply:
require 만으로 exception이 발생하기 때문에 try catch 블럭으로 한 번 더 잡을 필요는 없습니다.
아래 처럼 작성해 볼 수 있겠네요 🙂
```suggestion
        require(num != null && num >= 0) { "exception message" }
```